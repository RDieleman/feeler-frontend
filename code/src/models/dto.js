export function AddBookDTO(bookshelfId, isbn){
    this.bookshelfId = bookshelfId;
    this.isbn13 = isbn;
}

export function GetBookshelfDTO(bookshelfId){
    this.bookshelfId = bookshelfId;
}

export function RemoveBookDTO(bookshelfId, isbn){
    this.bookshelfId = bookshelfId;
    this.isbn13 = isbn;
}

export function UpdateBookDTO(bookshelfId, isbn, status){
    this.bookshelfId = bookshelfId;
    this.isbn13 = isbn;
    this.status = status;
}

