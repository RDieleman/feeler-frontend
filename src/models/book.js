//Book info
export function Book(title, overview, isbn, imageUrl){
    this.title = title;
    this.overview = (overview) ? overview : "No description found.";
    this.isbn = isbn;
    this.imageUrl = imageUrl;
}