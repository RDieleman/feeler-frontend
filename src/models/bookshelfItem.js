export function BookshelfItem(id, isbn, status){
    this.id = id;
    this.isbn = isbn;
    this.status = status;
}