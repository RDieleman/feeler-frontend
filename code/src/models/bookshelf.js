//Bookshelf
import {BookshelfItem} from "./bookshelfItem";

export function Bookshelf(id, content){
    this.id = id;
    this.content = [];
    content.forEach(i => {
        this.content.push(
            new BookshelfItem(
                i.id,
                i.isbn13,
                i.status
            )
        );
    })
    this.getRead = () =>{
        const items = [];
        this.content.forEach(i => {
            if (i.status === "READ") {
                items.push(i);
            }
        });
        return items;
    }

    this.getUnread = () =>{
        const items = [];
        this.content.forEach(i => {
            if (i.status === "UNREAD") {
                items.push(i);
            }
        });
        return items;
    }

    this.getReading = () =>{
        const items = [];
        this.content.forEach(i => {
            if (i.status === "READING") {
                items.push(i);
            }
        });
        return items;
    }
}