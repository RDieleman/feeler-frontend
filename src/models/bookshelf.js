//Bookshelf
import {BookshelfItem} from "./bookshelfItem";

export function Bookshelf(id, content){
    this.id = id;
    this.content = [];
    content.forEach(i => {
        this.content.push(
            new BookshelfItem(
                i.id,
                i.isbn,
                i.status
            )
        );
    })
}