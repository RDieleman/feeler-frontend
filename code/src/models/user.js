import {Bookshelf} from "./bookshelf";

export function User(id, name, bookshelf){
    this.id = id;
    this.name = name;
    this.bookshelf = new Bookshelf(
        bookshelf.id,
        bookshelf.content
    );
}