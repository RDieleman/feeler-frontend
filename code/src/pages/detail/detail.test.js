import React from "react";
import {shallow} from "enzyme";
import {Book} from "../../models/book";
import DetailPage from "./detail.page";

it("Expect to render Detail page", () => {
    expect(shallow(
        <DetailPage
            isbn="test"
        />)).toMatchSnapshot();
});