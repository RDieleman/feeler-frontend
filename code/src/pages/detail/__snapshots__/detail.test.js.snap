// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect to render Detail page 1`] = `ShallowWrapper {}`;
